src/email/mailchimp/audience_analysis.R

```diff
@@ -0,0 +1,563 @@
+box::use(
+  src/email/mailchimp/audience,
+  src/email/mailchimp/base_api
+)
+
+box::use(
+  dplyr,
+  gg = ggplot2,
+  gghdx,
+  httr2,
+  lubridate,
+  purrr,
+  rnaturalearth,
+  scales,
+  sf,
+  stringr,
+  tidyr
+)
+
+member_list <- audience$mc_members()
+df_interests <- audience$mc_groups() |>
+  dplyr$select(
+    interest_id,
+    interest = name,
+    group = title
+  )
+
+#' Get number of emails sent to subscriber
+#'
+#' Provided a `member_id` (subscriber_hash in the Mailchimp documentation sometimes),
+#' we pull in the number of mailings they have been sent.
+#'
+#' @export
+mc_email_count <- function(member_id) {
+  email_count <- mc_email_count_req(member_id = member_id)
+
+  offset <- 1000
+  while (email_count == offset) {
+    email_count <- email_count + mc_email_count_req(member_id = member_id, offset = offset)
+    offset <- offset + 1000
+  }
+  email_count
+}
+
+#' Repeatable call to use offset
+mc_email_count_req <- function(member_id, offset = 0) {
+  base_api$mc_api() |>
+    httr2$req_url_path_append(
+      "members",
+      member_id,
+      "activity-feed"
+    ) |>
+    httr2$req_url_query(
+      count = 1000,
+      offset = offset,
+      activity_filters = "sent"
+    ) |>
+    httr2$req_perform() |>
+    httr2$resp_body_json() |>
+    purrr$pluck("activity") |>
+    length()
+}
+
+# get information on members
+df_members <- purrr$map(
```

Comment:
i think this `df_members` is the dataset we want to pipeline. The rest of analysis can come from it